### PR TITLE
crd: strip leading yaml breaks

### PR DESCRIPTION
**Pull request by staebler**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -75,9 +76,10 @@ manifests: crd
 
 # Generate CRD yaml from our api types:
 .PHONY: crd
-crd: ensure-controller-gen
+crd: ensure-controller-gen ensure-yq
 	rm -rf ./config/crds
 	'$(CONTROLLER_GEN)' crd paths=./pkg/apis/hive/v1 output:dir=./config/crds
+	yq -y -i . ./config/crds/*.yaml
```
Comment (gregsheremeta) [inline Makefile:82]: how does this work? I don't see a mention of `-y` in the yq docs. Is it quiet mode? And does this just make yq do a reformat?

Comment (staebler) [reply Makefile:82]: The `-y` flag makes the output yaml instead of json.

Comment (staebler) [reply Makefile:82]: yq reads in the file then outputs it as yaml and replaces the file with the output.

Comment (gregsheremeta) [reply Makefile:82]: ok cool. I have to use `r` for read in the version I have. Shrug. lgtm!

Comment (csrwng) [reply Makefile:82]: The issue is there are multiple versions of yq out there that work slightly differently.

**Comment by staebler** [comment]
I added `ensure-yq` as a dependency to the `verify-crd` target.

I added a comment about the use of yq in the Makefile.

File: Makefile (hunk 1 above)
Comment (staebler) [reply Makefile:84]: Yes, you have a completely different yq.

Comment (staebler) [reply Makefile:84]: Ah, I am using the wrong one in the makefile.

**Pull request closed without merging**